public/app/plugins/panel/canvas/CanvasPanel.tsx

```diff
@@ -130,7 +168,60 @@ export class CanvasPanel extends Component<Props, State> {
     return changed;
   }
 
+  inlineEditButtonClick = () => {
+    if (isInlineEditOpen) {
+      this.forceUpdate();
+      this.setActivePanel();
+      return;
+    }
+
+    this.setActivePanel();
+    this.setState({ openInlineEdit: true });
+    isInlineEditOpen = true;
+  };
+
+  inlineEditButtonClose = () => {
+    this.setState({ openInlineEdit: false });
+    isInlineEditOpen = false;
+  };
+
+  setActivePanel = () => {
+    activeCanvasPanel = this;
+    activePanelSubject.next({ panel: this });
+  };
+
+  renderInlineEdit = () => {
+    return <InlineEdit onClose={() => this.inlineEditButtonClose()} />;
+  };
+
   render() {
-    return this.scene.render();
+    return (
+      <>
+        {this.scene.render()}
+        {this.props.options.inlineEditing && !this.isEditing && (
+          <div>
+            <div className={this.styles.inlineEditButton}>
+              <Button
+                size="lg"
+                variant="secondary"
+                icon="edit"
+                data-btninlineedit={this.props.id}
+                onClick={this.inlineEditButtonClick}
+              />
+            </div>
+            {this.state.openInlineEdit && this.renderInlineEdit()}
+          </div>
+        )}
+      </>
+    );
   }
 }
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
```

Comment:
Deprecated way of setting styles here, we can use the `useStyles2` hook in this file as well (similar feedback as to `InlineEdit` component)

Reply:
yes - but being a class component cannot use the hooks. And not sure if we want to switch..